cms/djangoapps/import_from_modulestore/tasks.py

```diff
@@ -0,0 +1,110 @@
+"""
+Tasks for course to library import.
+"""
+
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from django.db import transaction
+from edx_django_utils.monitoring import set_code_owner_attribute
+
+from openedx_learning.api import authoring as authoring_api
+from openedx_learning.api.authoring_models import LearningPackage
+from openedx.core.djangoapps.content_staging import api as content_staging_api
+
+from .constants import IMPORT_FROM_MODULESTORE_STAGING_PURPOSE
+from .data import ImportStatus
+from .helpers import get_items_to_import, ImportClient
+from .models import Import, PublishableEntityImport, StagedContentForImport
+from .validators import validate_composition_level
+
+log = get_task_logger(__name__)
+
+
+@shared_task
+@set_code_owner_attribute
+def save_legacy_content_to_staged_content_task(import_uuid: str) -> None:
+    """
+    Save courses to staged content task by sections/chapters.
+    """
+    try:
+        import_event = Import.objects.get(uuid=import_uuid)
+    except Import.DoesNotExist:
+        log.info('Import event not found for UUID %s', import_uuid)
+        return
+
+    import_event.clean_related_staged_content()
+    import_event.set_status(ImportStatus.STAGING)
+    try:
+        with transaction.atomic():
+            items_to_import = get_items_to_import(import_event)
+            for item in items_to_import:
+                staged_content = content_staging_api.stage_xblock_temporarily(
+                    item,
+                    import_event.user.id,
+                    purpose=IMPORT_FROM_MODULESTORE_STAGING_PURPOSE,
+                )
+                StagedContentForImport.objects.create(
+                    staged_content=staged_content,
+                    import_event=import_event,
+                    source_usage_key=item.location
+                )
+
+            if items_to_import:
+                import_event.set_status(ImportStatus.STAGED)
+            else:
+                import_event.set_status(ImportStatus.STAGING_FAILED)
+    except Exception as exc:  # pylint: disable=broad-except
+        import_event.set_status(ImportStatus.STAGING_FAILED)
+        raise exc
+
+
+@shared_task
+@set_code_owner_attribute
+def import_staged_content_to_library_task(
+    usage_keys_string: list[str],
```

Comment:
```suggestion
    usage_key_strings: list[str],
```
this is a list of strings, each of which is a usage key

usage_keys_string implies a single string made of many usage keys

Reply:
Changed.